### ISSUE TITLE
Opening stream of M4A file is throwing 'out of range' exception

### ISSUE THREAD

**Issue by kirk-marple** [Zeugma440/atldotnet#95]
## The problem

Opening stream of M4A file is throwing 'out of range' exception.  (File is zipped and attached.)

I've reproduced this on several M4A files via Zoom recordings, but works fine on other files.
[audio_only_004.zip](https://github.com/Zeugma440/atldotnet/files/6482293/audio_only_004.zip)

Also, opening this with a file path, not stream, works fine as well.

## Environment

        <PackageReference Include="z440.atl.core" Version="3.20.0" />

## Details

System.ArgumentOutOfRangeException
  HResult=0x80131502
  Message=Stream length must be non-negative and less than 2^31 - 1 - origin. (Parameter 'offset')
  Source=System.Private.CoreLib
  StackTrace:
   at System.IO.MemoryStream.Seek(Int64 offset, SeekOrigin loc)

## Code To Reproduce Issue [ Good To Have ]

             var track = new ATL.Track(stream, mimeType);

mimeType: audio/mp4

**Comment by Zeugma440** [comment]
Hello Kirk, and thanks for your detailed feedback.

I've just opened your sample file with a very simple code and it seems to work as expected.

```csharp
using (FileStream fs = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read))
{
     Track t = new Track(fs, "audio/mp4");
     Console.WriteLine("title : " + t.Title);
}
```

![image](https://user-images.githubusercontent.com/29414549/118390334-c6809b00-b62e-11eb-8daf-e5f299d39b77.png)

Here are a few pointers to potential issues :
- Is there something more you're using the stream for after creating the `Track` ?
- Is the stream you're using able to seek backwards ? Is it constrained by a buffer size ?
- If possible, could you send me a full stracktrace of your crash ?


Thanks ~

**Comment by kirk-marple** [comment]
Thanks for the quick reply.

I have this running in an XUnit test, and am copying the file down from HTTP URI into a memory stream.

Digging into this more, you can see the QuickWatch for the stream before the exception, and after the exception.  In the exception, you can see it's actually within MemoryStream.

![exception-quickwatch](https://user-images.githubusercontent.com/13594550/118409543-c63ccb80-b63f-11eb-922b-1e2ab821eeb3.png)
![stream-after-quickwatch](https://user-images.githubusercontent.com/13594550/118409544-c6d56200-b63f-11eb-96e2-0aa62a80cbad.png)
![stream-before-quickwatch](https://user-images.githubusercontent.com/13594550/118409545-c6d56200-b63f-11eb-9536-fce8460a766f.png)

```
        protected static async Task<(Stream Stream, string FileName, string MimeType)> GetStreamAsync(Uri uri)
        {
            string fileName = Path.GetFileName(uri.LocalPath);
            string mimeType = FormatHelpers.InferMimeType(Path.GetExtension(uri.LocalPath));

            var ms = new MemoryStream();
            using var stream = await Helpers.HttpClient.GetStreamAsync(uri);

            await stream.CopyToAsync(ms);

            ms.Position = 0L;

            return (ms, fileName, mimeType);
        }
```

**Comment by Zeugma440** [comment]
Finally got it !

The culprit was a sinister cast issue that only happened when using a `MemoryStream` (probably because `MemoryStream.Seek` doesn't do the same checks as `FileStream.Seek`)

**Comment by Zeugma440** [comment]
Fix published in today's release (v3.21)
